### DISCUSSION TITLE
Android app not showing updates to UI

### DISCUSSION THREAD

**Discussion by connectMyClub** [ionic-team/capacitor#7626] [Q&A]
Hi all,

_Ionic, Capacitor, Angular app_

Recently our Android app updates are not updating the user's app without first clearing app's cache & storage. There's no issue with our iOS app

The Android user can update their app from the Play Store. The version number increments as expected but what is shown in the view is not always the latest version. Some users are not having this issue.

The PWA appears to be rolling back to the previous, I assume, cached versions. You can Shift + Ctrl + R to get the latest version but then an F5 refresh rolls back the PWA to a previous version.

Does anyone have and ideas why this could be happening?

**Comment by rbalet** [comment]
@connectMyClub Did you find any solution ? 

It does indeed seems to be a caching problem. 

I'm actually having the same problem, while I've not managed yet to fix it, I'll post here what I've tried in ways to help you and others.  
Maybe together we'll find a solution :)

So, cache can be handled by the service worker.
Here's an example with angular 

1. Add service worker -> `ng add @angular/pwa`
2. In your `main.ts` inside the `providers
```typescript
provideServiceWorker('ngsw-worker.js', {
  enabled: !isDevMode(),
  registrationStrategy: 'registerWhenStable:30000',
}),
```
3. in your `app.component.ts` once your app is ready 
```typescript

private _setServiceWorker() {
  this._swUpdate.checkForUpdate()
  
  this._swUpdate.unrecoverable.subscribe(() => {
    window.location.reload()
  })

  this._swUpdate.versionUpdates.pipe(first()).subscribe((versionEvent) => {
    if (
      versionEvent.type === 'VERSION_READY' &&
      versionEvent.currentVersion.hash !== versionEvent.latestVersion.hash
    ) {
      this._swUpdate.activateUpdate().then(() => {
        // @ToDo: see if warning is required
        window.location.reload()
      })
    }
  })
}
```
